### PR TITLE
Issue 42 validate hashes

### PR DESCRIPTION
**Pull request by yeraydavidrodriguez**
This will fix issue #42 : Validate hashes values in the CommitHash and ShortCommitHash objects.

Aditionally, it will fix the parsing of a text commit (commit-subject-parser:getJobRef) to return without throwing an exception when no job ref is found in the subject.

### CODE REVIEW

File: src/errors.ts [hunk 1]
```diff
@@ -76,3 +76,17 @@ export class MissingNewJobMessageError extends Error {
...
+  constructor(hash: string) {
+    super(`Invalid SHA-1 commit hash: ${hash}`)
+    Object.setPrototypeOf(this, MissingQueueNameInCommitSubjectError.prototype)
+  }
+}
+
+export class InvalidShortHash extends Error {
+  constructor(hash: string) {
+    super(`Invalid 7-characters SHA-1 commit hash: ${hash}`)
+    Object.setPrototypeOf(this, MissingQueueNameInCommitSubjectError.prototype)
```
Comment (josecelano) [inline src/errors.ts:90]: `MissingQueueNameInCommitSubjectError` should be `InvalidShortHash` @yeraydavidrodriguez

**Review by josecelano — CHANGES_REQUESTED** [review]

File: src/commit-hash.ts [hunk 2]
```diff
@@ -7,10 +8,18 @@ export class CommitHash implements Nullable {
...
 
   constructor(value: string) {
-    // TODO: validation
+    if (value !== NO_COMMIT_HASH) {
+      this.validateHash(value)
+    }
     this.value = value
   }
 
+  validateHash(value: string): void {
```
[the remark this answers is not in the document]
Comment (yeraydavidrodriguez) [reply src/commit-hash.ts:17]: I think that is a good convention to follow. I will make necessary changes.

File: src/errors.ts (hunk 1 above)
Comment (yeraydavidrodriguez) [reply src/errors.ts:90]: Ups. Sorry about that.

File: src/errors.ts [hunk 3]
```diff
@@ -76,3 +76,17 @@ export class MissingNewJobMessageError extends Error {
     Object.setPrototypeOf(this, MissingNewJobMessageError.prototype)
   }
 }
+
+export class InvalidHash extends Error {
+  constructor(hash: string) {
+    super(`Invalid SHA-1 commit hash: ${hash}`)
+    Object.setPrototypeOf(this, MissingQueueNameInCommitSubjectError.prototype)
```
[the remark this answers is not in the document]
Comment (yeraydavidrodriguez) [reply src/errors.ts:83]: Yep, sorry about that.

File: src/short-commit-hash.ts [hunk 4]
```diff
@@ -9,10 +10,18 @@ export class ShortCommitHash implements Nullable {
...
 
   constructor(value: string) {
-    // TODO: validation
+    if (value !== NO_SHORT_COMMIT_HASH) {
+      this.validateHash(value)
+    }
     this.value = value
   }
 
+  validateHash(value: string): void {
```
[the remark this answers is not in the document]
Comment (yeraydavidrodriguez) [reply src/short-commit-hash.ts:19]: FIxed.

File: src/errors.ts [hunk 5]
```diff
@@ -76,3 +76,17 @@ export class MissingNewJobMessageError extends Error {
     Object.setPrototypeOf(this, MissingNewJobMessageError.prototype)
   }
 }
+
+export class InvalidHash extends Error {
```
Comment (josecelano) [inline src/errors.ts:80]: @yeraydavidrodriguez all errors have the suffix `Error`

File: src/errors.ts [hunk 6]
```diff
@@ -76,3 +76,17 @@ export class MissingNewJobMessageError extends Error {
...
+  constructor(hash: string) {
+    super(`Invalid SHA-1 commit hash: ${hash}`)
+    Object.setPrototypeOf(this, MissingQueueNameInCommitSubjectError.prototype)
+  }
+}
+
+export class InvalidShortHash extends Error {
```
Comment (josecelano) [inline src/errors.ts:87]: @yeraydavidrodriguez all errors have the suffix `Error`

**Review by josecelano — APPROVED** [review]

File: src/errors.ts (hunk 5 above)
Comment (yeraydavidrodriguez) [reply src/errors.ts]: You are right. Fixed.

File: src/errors.ts (hunk 6 above)
Comment (yeraydavidrodriguez) [reply src/errors.ts]: Fixed.

**Review by josecelano — APPROVED** [review]

**Pull request merged**